src/urbit/app/chess.hoon

```diff
@@ -273,13 +273,52 @@
       :-  :~  :*  %give  %fact  ~[/challenges]
                   %chess-update  !>([%challenge src.bowl challenge])
               ==
+              :*  %pass   /hark-store
+                  %agent  [our.bowl %hark-store]
+                  %poke   :-  %hark-action
+                          !>  :-  %unread-count
+                              :-  %chess
+                              :-  /challenges
+                              :-  %.y  1
+              ==
+              :*  =/  title=(list content:hark)
+                    ~[[%text (crip "{<src.bowl>} has sent a challenge")]]
+                  =/  body=(list content:hark)
+                    ~[[%text event.challenge]]
+                  =/  =bin:hark
+                    [/chess [%chess /challenges]]
+                  =/  hark-action=action:hark
+                    ::  XX: should link to game
+                    [%add-note bin title body now.bowl / /chess]
```

Comment:
It looks like the contents of the `link` field get passed as a URL variable with name "grid-note", so linking to a game (or even the games/challenges menu) just involves parsing the value from the URL variable and setting the state accordingly on the frontend.

If you're feeling adventurous, this is also an opportunity to make all invalid URLs redirect to the base URL (`/apps/chess/`).

Reply:
Good thinking. I think this should wait until the new Landscape notifications are implemented - what if our links are incompatible with Nu-Landscape? - so I've opened a new issue for it.